### PR TITLE
feat(auth): Invite and register process

### PR DESCRIPTION
**Pull request by ThomasKro**
- implemented user invite process
- implemented send mail to invited user
- implemented register for new user (currenlty needs token)
- fixed tenant issues

### CODE REVIEW

**Comment by ThomasKro** [comment]
@beyer-johannes updated the invite process, now you can review

File: packages/antify-admin/src/server/api/global/tenants.get.ts [hunk 1]
```diff
@@ -1,43 +1,46 @@
...
-    const guard = useGuard(useAuthorizationHeader(event));
-    const tenantId = useTenantHeader(event);
+  // TODO:: use middleware
+  const guard = useGuard(useAuthorizationHeader(event));
+  const tenantId = useTenantHeader(event);
 
-    if (!guard.hasPermissionTo(PermissionId.CAN_READ_TENANT, tenantId)) {
-        throw new HttpForbiddenError();
-    }
+  // TODO:: User should always have read access to tenants he is assigned to.
```
Comment (beyer-johannes) [inline packages/antify-admin/src/server/api/global/tenants.get.ts:17]: gruad.hasPermissionTo check this already

File: packages/antify-admin/src/server/datasources/auth/migrations/20220729052805_init/migration.sql [hunk 2]
```diff
@@ -12,7 +12,7 @@ CREATE TABLE `Tenant` (
 CREATE TABLE `User` (
     `id` VARCHAR(191) NOT NULL,
     `email` VARCHAR(191) NOT NULL,
-    `password` VARCHAR(191) NOT NULL,
+    `password` VARCHAR(191) NULL DEFAULT '',
```
Comment (beyer-johannes) [inline packages/antify-admin/src/server/datasources/auth/migrations/20220729052805_init/migration.sql:15]: Why?

**Review by beyer-johannes — CHANGES_REQUESTED** [review]

File: packages/antify-admin/src/server/datasources/auth/migrations/20220729052805_init/migration.sql (hunk 2 above)
Comment (ThomasKro) [reply packages/antify-admin/src/server/datasources/auth/migrations/20220729052805_init/migration.sql:15]: Because

**Review by beyer-johannes — APPROVED** [review]

**Pull request merged**